### ISSUE TITLE
Failed to connect, are you sure your server works via FTP or FTPS? Users sometimes get this error when the server only supports SFTP.

### ISSUE THREAD

**Issue by bhekor** [SamKirkland/FTP-Deploy-Action#402]
I'm using Namecheap and it was working perfectly before now. I have disabled Proxied on Cloudflare, because I'm using Cloudflare.

Run SamKirkland/FTP-Deploy-Action@v4.3.4
----------------------------------------------------------------
🚀 Thanks for using ftp-deploy. Let's deploy some stuff!   
----------------------------------------------------------------
If you found this project helpful, please support it
by giving it a ⭐ on Github --> https://github.com/SamKirkland/FTP-Deploy-Action
or add a badge 🏷️ to your projects readme --> https://github.com/SamKirkland/FTP-Deploy-Action#badge
Failed to connect, are you sure your server works via FTP or FTPS? Users sometimes get this error when the server only supports SFTP.

----------------------------------------------------------------
--------------  🔥🔥🔥 an error occurred  🔥🔥🔥  --------------
----------------------------------------------------------------

----------------------------------------------------------------
----------------------  full error below  ----------------------
----------------------------------------------------------------

Error: Timeout (control socket)
    at Socket.<anonymous> (/home/runner/work/_actions/SamKirkland/FTP-Deploy-Action/v4.3.4/dist/index.js:5288:33)
    at Object.onceWrapper (node:events:641:28)
    at Socket.emit (node:events:527:28)
    at Socket._onTimeout (node:net:5[16](https://github.com/username/app/actions/runs/5106021444/jobs/9177966952#step:9:17):8)
    at listOnTimeout (node:internal/timers:559:[17]((https://github.com/username/app/actions/runs/5106021444/jobs/9177966952#step:9:18))
    at processTimers (node:internal/timers:502:7)
Error: Error: Timeout (control socket)

**Comment by xannasavin** [comment]
I have the very same issue

```
Error: Timeout (control socket)
    at Socket.<anonymous> (/home/runner/work/_actions/SamKirkland/FTP-Deploy-Action/v4.3.4/dist/index.js:5288:33)
    at Object.onceWrapper (node:events:641:28)
    at Socket.emit (node:events:5[27](https://github.com/xannasavin/YOURLS/actions/runs/5132500397/jobs/9233873454#step:3:28):[28](https://github.com/xannasavin/YOURLS/actions/runs/5132500397/jobs/9233873454#step:3:29))
    at Socket._onTimeout (node:net:516:8)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7)
Error: Error: Timeout (control socket)
```

**Comment by thechandanbhagat** [comment]
Any Updates?  I am also facing the same issue
![2023-06-05,Mon,15-45-23](https://github.com/SamKirkland/FTP-Deploy-Action/assets/6848264/fa3d6e1a-d7a2-41f9-82b9-0e24b4d1ca12)

**Comment by AffiTheCreator** [comment]
same here

**Comment by dersonsena** [comment]
I was having this same problem and saw in [another issue](https://github.com/SamKirkland/FTP-Deploy-Action/issues/239) that someone else was having problems with special characters or spaces in the password.

My solution was: I created a password with **only letters and numbers** and the problem no longer occurred, try that.

**Comment by xannasavin** [comment]
I updated my password to alphanumerical, but it did not solve the issue

**Comment by xannasavin** [comment]
```yaml
# https://github.com/marketplace/actions/ftp-deploy
# 08.03.2023

name: 🚀 Deploy website on push

# Controls when the action will run.
on:
  # Triggers the workflow on push or pull request events but only for the master branch
  push:
    branches: [ main ]

  # Allows you to run this workflow manually from the Actions tab
  workflow_dispatch:

# A workflow run is made up of one or more jobs that can run sequentially or in parallel
jobs:
  web-deploy:
    name: 🎉 Deploy
    runs-on: ubuntu-latest
    steps:
      - name: 🚚 Get latest code
        uses: actions/checkout@v3.4.0

      # https://github.com/actions/checkout
      - name: Checkout WP Plugin
        uses: actions/checkout@v3.4.0
        with:
          repository: XXX
          token: ${{ secrets.my_pat }}
          path: mu-plugins

      - name: 📂 Sync files
        uses: SamKirkland/FTP-Deploy-Action@4.3.3
        with:
          server: ${{ secrets.FTP_SERVER }}
          username: ${{ secrets.FTP_USER }}
          password: ${{ secrets.FTP_PASSWORD }}
          exclude: |
            **/.git*
            **/.git*/**
            .git*
            .git*/**
            .github/*
           [...]
```

```
Run SamKirkland/FTP-Deploy-Action@4.3.3
----------------------------------------------------------------
🚀 Thanks for using ftp-deploy. Let's deploy some stuff!   
----------------------------------------------------------------
If you found this project helpful, please support it
by giving it a ⭐ on Github --> https://github.com/SamKirkland/FTP-Deploy-Action
or add a badge 🏷️ to your projects readme --> https://github.com/SamKirkland/FTP-Deploy-Action#badge
Failed to connect, are you sure your server works via FTP or FTPS? Users sometimes get this error when the server only supports SFTP.

----------------------------------------------------------------
--------------  🔥🔥🔥 an error occurred  🔥🔥🔥  --------------
----------------------------------------------------------------

----------------------------------------------------------------
----------------------  full error below  ----------------------
----------------------------------------------------------------

Error: Timeout (control socket)
    at Socket.<anonymous> (/home/runner/work/_actions/SamKirkland/FTP-Deploy-Action/4.3.3/dist/index.js:5288:58)
    at Object.onceWrapper (node:events:6[41](https://github.com/xannasavin/fuehringer/actions/runs/5209528020/jobs/9399480241#step:4:42):28)
    at Socket.emit (node:events:527:28)
    at Socket._onTimeout (node:net:516:8)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:[50](https://github.com/xannasavin/fuehringer/actions/runs/5209528020/jobs/9399480241#step:4:51)2:7)
Error: Error: Timeout (control socket)
```

**Comment by IcyFoxe** [comment]
Maybe this will help someone:
Check if you don't have some permission restrictions on your server host.
In my case FTP could be accessed only from EU countries. Once I turned that off, everything started to work correctly.

**Comment by bhekor** [comment]
If you are using Namecheap or basically checking your hosting platform, it's possible the server (host) on the FTP credential is probably different from your domain name, as mine domain is an addon domain and it's using the hosting primary domain.

**Comment by bhekor** [comment]
If you are using Namecheap or basically checking your hosting platform, it's possible the server (host) on the FTP credential is probably different from your domain name, as mine domain is an addon domain and it's using the hosting primary domain.

I want to close as I've resolved this.

**Comment by xannasavin** [comment]
sorry, this is unfortunately not resolved...
Changing password did not help.
There is no restriction and also the server is correct - this worked fine in the past.

**Comment by xannasavin** [comment]
I can no re-open?!

**Comment by xannasavin** [comment]
hmm... it looks like it's working again - will comment once more if not.

**Comment by Arman091** [comment]
Yes!! I GOT IT ,Problem was with secret Keys i was providing secret keys as "keyvalue" also i changed my password to alphanumeric chracter and no special character because github action throws error if you provide any secret key with special characters i think that's it.

1.give username no special charachter just plain text while creating new ftp account

**Comment by gedeonagmas** [comment]
i have got the same problem finally i solved it by replace the secrets.FTP_SERVER by your hosting or server ip address not domain name.

**Comment by Okteeicode** [comment]
I was having same use, I am using name cheap , i created a new NEW FTP account trying to connect to that i keep getting error,i then used my Cpanel login credential and its work

**Comment by Aymsep** [comment]
just resolved the issue , all i did is just delete the ftp account , and create new one .

**Comment by ewald-ai** [comment]
If you are using Cloudflare in your primary domain, it may block requests from GitHub actions. So, I solved the problem by bypassing Cloudflare (Using the server IP address instead of the domain), and this works for me.

**Comment by destiny2222** [comment]
> If you are using Namecheap or basically checking your hosting platform, it's possible the server (host) on the FTP credential is probably different from your domain name, as mine domain is an addon domain and it's using the hosting primary domain.
> 
> I want to close as I've resolved this.

how do you resolve it, i am having the same issue here

**Comment by mugambbo** [comment]
I added  "sftp_only: true" and it worked!
`with:
          server: ${{ secrets.FTP_HOST }}
          username: ${{ secrets.FTP_USERNAME }}
          password: ${{ secrets.FTP_PASSWORD }}
          protocol: ftps`

**Comment by Victorvalour** [comment]
I got the same problem. My issue was because I changed the nameservers to a different one. When I changes it back it started working again.

**Comment by MadalitsoNyemba** [comment]
> I added "sftp_only: true" and it worked!
> 
> ```
> with:
>           server: ${{ secrets.FTP_HOST }}
>           username: ${{ secrets.FTP_USERNAME }}
>           password: ${{ secrets.FTP_PASSWORD }}
>           protocol: ftps
> ```

This is what worked for me.

**Comment by guruhowtodo** [comment]
Me too faced the same kind of issue, tried changing many things and following lots of article nothing worked out. accidently changed the password for FTP account for easy accessing without any special character after running its charm things are worked smoothly.
